services/movie_session.py

```diff
@@ -0,0 +1,48 @@
+from db.models import MovieSession
+from django.utils.dateparse import parse_date
+from datetime import datetime
+
+
+def create_movie_session(
+        movie_show_time: str | datetime,
+        movie_id: int,
+        cinema_hall_id: int
+) -> None:
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie_id=movie_id,
+        cinema_hall_id=cinema_hall_id
+    )
+
+
+def get_movies_sessions(session_date: str | None = None) -> list[MovieSession]:
+    queryset = MovieSession.objects.all()
+    if session_date:
+        queryset = queryset.filter(show_time__date=parse_date(session_date))
+    return queryset
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: datetime | None = None,
+        movie_id: int | None = None,
+        cinema_hall_id: int | None = None
+) -> MovieSession:
+    session = MovieSession.objects.get(id=session_id)

```

Comment:
Consider handling the `MovieSession.DoesNotExist` exception in `update_movie_session` to avoid potential errors when a session with the given ID does not exist.